xbmc/addons/interfaces/gui/General.cpp

```diff
@@ -212,6 +215,37 @@ ADDON_HARDWARE_CONTEXT Interface_GUIGeneral::get_hw_context(KODI_HANDLE kodiBase
   return CServiceBroker::GetWinSystem()->GetHWContext();
 }
 
+int Interface_GUIGeneral::get_adjust_refresh_rate(KODI_HANDLE kodiBase)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "kodi::gui::{} - invalid data", __func__);
+    return 0;
+  }
+
+  switch (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+      CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE))
+  {
+    case AdjustRefreshRate::ADJUST_REFRESHRATE_OFF:
+      return kodi::gui::AdjustRefreshRateStatus::ADJUST_REFRESHRATE_STATUS_OFF;
+      break;
+    case AdjustRefreshRate::ADJUST_REFRESHRATE_ON_START:
+      return kodi::gui::AdjustRefreshRateStatus::ADJUST_REFRESHRATE_STATUS_ON_START;
+      break;
+    case AdjustRefreshRate::ADJUST_REFRESHRATE_ON_STARTSTOP:
+      return kodi::gui::AdjustRefreshRateStatus::ADJUST_REFRESHRATE_STATUS_ON_STARTSTOP;
+      break;
+    case AdjustRefreshRate::ADJUST_REFRESHRATE_ALWAYS:
+      return kodi::gui::AdjustRefreshRateStatus::ADJUST_REFRESHRATE_STATUS_ALWAYS;
+      break;
+    default:
+      CLog::Log(LOGERROR, "kodi::gui::{} - Unhandled Adjust refresh rate setting", __func__);
+      return kodi::gui::AdjustRefreshRateStatus::ADJUST_REFRESHRATE_STATUS_OFF;
```

Comment:
C++ related add-on headers should never be used inside Kodi itself (currently are they still only used by `#include "..."` in Kodi to confirm they are on compile OK).

Reply:
what type of change you are asking here?

Reply:
He means the CLog::Log call, you need to use the equivalent from the binary interface I think

Reply:
To have the enum within "C" header, to use a C++ namespace within C ABI between add-on and Kodi is not allowed 😉.